src/bin/lockin.rs

```diff
@@ -256,7 +260,11 @@ const APP: () = {
                         }
                         Conf::InPhase => output.re >> 16,
                         Conf::Quadrature => output.im >> 16,
-                        Conf::Modulation => DAC_SEQUENCE[i] as i32,
+
+                        // Note: Because the signal generator has a period equal to one batch size,
+                        // it's okay to only update it when outputting the modulation waveform, as
+                        // it will perfectly wrap back to zero phase for each batch.
+                        Conf::Modulation => signal_generator.next() as i32,
```

Comment:
I've observed that the waveform appears out-of-order on the DAC outputs if the period is e.g. 3x batch size. This appears only at the very start and is rectified after a number of seconds. I am not yet aware of the cause.